_ml-commons-plugin/semantic-search.md

```diff
@@ -0,0 +1,1030 @@
+    "excludes": [
+      "passage_embedding"
+    ]
+  },
+  "query": {
+    "neural": {
+      "passage_embedding": {
+        "query_text": "wild west",
+        "model_id": "aVeif4oB5Vm0Tdw8zYO2",
+        "k": 5
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The results contain all five documents. The document order is now closer to the desired one because semantic meaning is considered:
+
+<details closed markdown="block">
+  <summary>
+    Results
+  </summary>
+  {: .text-delta}
+
+```json
+{
+  "took": 25,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 5,
+      "relation": "eq"
+    },
+    "max_score": 0.01585195,
+    "hits": [
+      {
+        "_index": "my-nlp-index",
+        "_id": "4",
+        "_score": 0.01585195,
+        "_source": {
+          "text": "A man who is riding a wild horse in the rodeo is very near to falling off .",
+          "id": "4427058951.jpg"
+        }
+      },
+      {
+        "_index": "my-nlp-index",
+        "_id": "2",
+        "_score": 0.015748845,
+        "_source": {
+          "text": "A wild animal races across an uncut field with a minimal amount of trees.",
+          "id": "1775029934.jpg"
+        }
+      },
+      {
+        "_index": "my-nlp-index",
+        "_id": "5",
+        "_score": 0.015177963,
+        "_source": {
+          "text": "A rodeo cowboy , wearing a cowboy hat , is being thrown off of a wild white horse .",
+          "id": "2691147709.jpg"
+        }
+      },
+      {
+        "_index": "my-nlp-index",
+        "_id": "1",
+        "_score": 0.013272902,
+        "_source": {
+          "text": "A West Virginia university women 's basketball team , officials , and a small gathering of fans are in a West Virginia arena .",
+          "id": "4319130149.jpg"
+        }
+      },
+      {
+        "_index": "my-nlp-index",
+        "_id": "3",
+        "_score": 0.011347735,
+        "_source": {
+          "text": "People line the stands which advertise Freemont 's orthopedics , a cowboy rides a light brown bucking bronco .",
+          "id": "2664027527.jpg"
+        }
+      }
+    ]
+  }
+}
+```
+</details>
+
+### Search with a combined keyword search and neural search
+
+To combine keyword search and neural search, you need to set up a [search pipeline]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/index/) that is similar to the ingest pipeline but runs at search time instead of ingestion time. The search pipeline you'll configure intercepts search results at an intermediate stage and applies the [`normalization_processor`]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/normalization-processor/) to them. The `normalization_processor` normalizes and combines the document scores from multiple query clauses, rescoring the documents according to the chosen normalization and combination techniques. 
```

Comment:
Nit, the mention of ingest pipeline made me have to read this sentence three times. Consider simplifying. 

```suggestion
To combine keyword search and neural search, you need to set up a [search pipeline]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/index/) that runs at search time instead of ingestion time. The search pipeline you'll configure intercepts search results at an intermediate stage and applies the [`normalization_processor`]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/normalization-processor/) to them. The `normalization_processor` normalizes and combines the document scores from multiple query clauses, rescoring the documents according to the chosen normalization and combination techniques. 
```

Reply:
Reworded.